### PR TITLE
GenericModelBinder integration tests part 1

### PR DESCRIPTION
**Pull request by rynowak**
- Covers simple scenario for each model binder.
- Covers scenarios mixing a generic model binder with a greedy model
  binder.

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.IntegrationTests/KeyValuePairModelBinderIntegrationTest.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            httpContext.Request.QueryString = new QueryString("?");
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(httpContext);
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.Null(modelBindingResult);
```
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/KeyValuePairModelBinderIntegrationTest.cs]: Should this be `Assert.NotNull` ?

**Comment by harshgMSFT** [comment]
:shipit: after addressing comment.

**Comment by rynowak** [comment]
> :shipit: after addressing comment.

Please be a gentleman and merge your changes so I can rebase on top :+1:

**Comment by harshgMSFT** [comment]
Merged.

**Pull request closed without merging**